### ISSUE TITLE
In Workspace Or Tasks.json, Allow For Non-Existent Settings ("config:<notFound>").

### ISSUE THREAD

**Issue by vindicatorr** [microsoft/vscode#202969]
Allow for any "not found" config setting to return empty instead of error out.
```
settings: {
  "anythingButSomethingA": "<some val>"
},
"tasks": {
    "version": "2.0.0",
    "options": {
        "env": {
            "SOMETHING_A": "${config:SomethingA}"
        }
    },
"tasks": [
      {
        "label": "Task 0",
        "type": "process",
        "options": {
            "cwd": "${workspaceFolder:<workspaceFolderName0>}",
            "env": {
            "SOMETHING_A": ""
            }
        },
        "command": "<script to set config values in this file>",
      },
      {
        "label": "Task 1",
        "type": "process",
        "options": {
            "cwd": "${workspaceFolder:<workspaceFolderName1>}"
        },
        "command": "make $SOMETHING_A",
      },
    <...>
  ]
}
```
`Variable ${config:SomethingA} can not be resolved because setting 'SomethingA not found.`

Note that with the initializing `Task 0`, I manually override the global environment variable, while any subsequent task will have that global variable set after `Task 0` is run, without having to copy/paste the `options.env` to each task, and it'll still error-out as the config setting "not found".

Now, I normally would have just had the script set the environment variables instead of the settings `config` variables, but the `"shellCommand.execute"` extension won't recognize those: https://github.com/augustocdias/vscode-shell-command/pull/19#issuecomment-1902904006.

EDIT0: ACK! After I started to copy/paste, I noticed I tested the override on the wrong task. The override DOES work as expected.

EDIT1: Nah, the request still stands. I just encountered a scenario where I need to run a task that is dependent on the environment variable being set via the config setting, and it just says that "not found" error message.
I have that task "dependsOn" another task that would have tested if that environment variable was set, and if not, emit a nicer message telling the user to run the environment/config-setting task first.

**Comment by meganrogge** [comment]
We're not going to do this as it's working as designed.

**Comment by vindicatorr** [comment]
"working as designed", but I've demonstrated the issue where it ought to be up to the .json writer to determine what happens when a config/setting/variable has a value or not.

For example, I can use `${env:DOESNTEXIST}` for an environment variable that doesn't exist and no error will be thrown by vscode.
If, as a .json/script writer, I need that value, I can inform them that the value is missing and how they can set it.
If using `${input:notYetMade}` is used for an input that doesn't exist, an error will be thrown by vscode like a missing `config` element.

I see it as contrary designs when 2 will throw an error while the other will be silent (which as I've stated, I would prefer to occur).

While I haven't written an extension yet, I do wonder what an extension would have to contend with when looking up a config key that may be missing. Would it be passed a `null` value which it can test for or will vscode throw an error that the extension would have to `try/catch`? No `try/catch` is available to the .json writer as the error is thrown before the script/command is called.
